packages/notifications/src/PushNotification/utils.ts

```
@@ -34,59 +38,87 @@ export const normalizeNativePermissionStatus = (
+	if (nativeMessage?.payload) {
+		normalized = normalizeFcmMessage(nativeMessage);
 	}
-	return null;
-};
-
-const getContent = (payload): PushNotificationMessage['content'] => {
-	const { action, body, imageUrl, title } = payload;
-	const apnsOptions = getApnsOptions(payload);
-	const fcmOptions = getFcmOptions(payload);
+	if (!normalized) {
+		return null;
+	}
+	const { body, imageUrl, title, action, options, data } = normalized;
 	return {
 		...(body && { body }),
 		...(imageUrl && { imageUrl }),
 		...(title && { title }),
-		...getAction(action),
-		...(!isEmpty(apnsOptions) && { apnsOptions }),
-		...(!isEmpty(fcmOptions) && { fcmOptions }),
+		...action,
+		...options,
+		...(!isEmpty(data) && { data }),
 	};
 };
 
-const getAction = (
-	action
-): Pick<PushNotificationMessage['content'], 'goToUrl' | 'deeplinkUrl'> => {
-	if (action[URL_ACTION]) {
-		return { goToUrl: action[URL_ACTION] };
+const normalizeApnsMessage = (apnsMessage): NormalizedValues => {
+	const { aps = { alert: {} }, data = {} } = apnsMessage;
+	const { body, title } = aps.alert;
+	// We want to eventually move provider specifics into the provider somehow, but without another provider use case to
+	// develop against, any decisions made regarding how to transform this data would likely be discarded. For now, we
+	// should not over-complicate this transformer
+	const action = getApnsAction(data.pinpoint) ?? {};
+	const imageUrl = data['media-url'];
+	const options = getApnsOptions(apnsMessage);
+	return { body, imageUrl, title, action, options, data };
+};
+
+const normalizeFcmMessage = (fcmMessage): NormalizedValues => {
+	try {
+		const payload = JSON.parse(fcmMessage.payload);
+		const { body, imageUrl, rawData: data, title } = payload;
+		const action = getFcmAction(payload.action) ?? {};
+		const options = getFcmOptions(payload);
+		return { body, imageUrl, title, action, options, data };
+	} catch (e) {
+		logger.error('An error ocurred parsing the native payload');
```

Nit: I think I'd prefer to let this bubble up and handle errors uniformly in `normalizeNativeMessage`

---

This error is only relevant for our FCM messages since they're coming in as a JSON string